public/scripts/socket-heroku.js

```diff
@@ -0,0 +1,45 @@
+/**
+ * Client-side Event Handling for deploy/heroku
+ */
+$(function () {
+  var socket = io();
+
+  $("#herokuAppName").keyup(function () {
+    validation.validateHerokuAppName($("#herokuAppName").val());
```

Comment:
It how templating works here. :)
Let's start by taking a deep breath.
there is a `layout.jade` file that calls other template files, like `deploy/heroku.jade` through the `block content` statement.

If you notice, `layout.jade` calls the script `script(src="/scripts/validation.js")`  and `heroku.jade` calls the script `script(src="/scripts/socket-heroku.js")`

So you have the output at deploy/heroku.html which includes both the scripts. Since validation.js is called before socket-heroku.js, there is no need to require it